src/hostnet/lib/host_uwt.ml

```diff
@@ -82,11 +82,14 @@ module Sockets = struct
         loop () in
       loop ()
 
-    let input ~reply ~src:(src, src_port) ~dst:(dst, dst_port) ~payload =
+    let input ?desc ~reply ~src:(src, src_port) ~dst:(dst, dst_port) ~payload () =
       (if Hashtbl.mem table (src, src_port, dst, dst_port) then begin
           Lwt.return (Some (Hashtbl.find table (src, src_port, dst, dst_port)))
         end else begin
-         let description = Ipaddr.to_string src ^ ":" ^ (string_of_int src_port) ^ "-" ^ (Ipaddr.to_string dst) ^ ":" ^ (string_of_int dst_port) in
+         let base_description = Ipaddr.to_string src ^ ":" ^ (string_of_int src_port) ^ "-" ^ (Ipaddr.to_string dst) ^ ":" ^ (string_of_int dst_port) in
```

Comment:
`String.concat` -> Sure.

Moving to the log -> The description is actually used in the record too:
```
           end else begin
             let last_use = Unix.gettimeofday () in
             let flow = { description; fd; last_use; reply} in
```
and then used from a bunch different Log calls (half a dozen here and the rest elsewhere e.g. on teardown). I think the correct extension of what you suggest is to turn `description` from a `string` into a `() -> string` and have the various logging locations evaluate it lazily, is that right?